src/main/java/ch/njol/skript/effects/EffStopSound.java

```diff
@@ -36,97 +28,109 @@
+				"stop sound[s] %strings% [(in|from) %-soundcategory%] [(from playing to|for) %players%]",
+				"stop playing sound[s] %strings% [(in|from) %-soundcategory%] [(to|for) %players%]");
 	}
-
-	@SuppressWarnings("null")
+	
+	@SuppressWarnings("NotNullFieldNotInitialized")
 	private Expression<String> sounds;
-	@Nullable
-	private Expression<SoundCategory> category;
-	@SuppressWarnings("null")
+	private @Nullable Expression<SoundCategory> category;
+	@SuppressWarnings("NotNullFieldNotInitialized")
 	private Expression<Player> players;
+	
+	private boolean allSounds;
 
 	@Override
-	@SuppressWarnings({"unchecked", "null"})
+	@SuppressWarnings({"unchecked"})
 	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
-		sounds = (Expression<String>) exprs[0];
-		if (SOUND_CATEGORIES_EXIST) {
+		allSounds = matchedPattern == 2;
+		if (allSounds) {
+			category = (Expression<SoundCategory>) exprs[0];
+			players = (Expression<Player>) exprs[1];
+		} else {
+			sounds = (Expression<String>) exprs[0];
 			category = (Expression<SoundCategory>) exprs[1];
 			players = (Expression<Player>) exprs[2];
-		} else {
-			players = (Expression<Player>) exprs[1];
 		}
+		
 		return true;
 	}
 
 	@Override
-	protected void execute(Event e) {
-		Object category = null;
-		if (SOUND_CATEGORIES_EXIST) {
-			category = SoundCategory.MASTER;
-			if (this.category != null) {
-				category = this.category.getSingle(e);
-				if (category == null)
-					return;
-			}
-		}
-		for (String sound : sounds.getArray(e)) {
-			Sound soundEnum = null;
-			try {
-				soundEnum = Sound.valueOf(sound.toUpperCase(Locale.ENGLISH));
-			} catch (IllegalArgumentException ignored) {}
-			if (soundEnum == null) {
-				if (SOUND_CATEGORIES_EXIST) {
-					for (Player p : players.getArray(e))
-						p.stopSound(sound, (SoundCategory) category);
-				} else {
-					for (Player p : players.getArray(e))
-						p.stopSound(sound);
-				}
+	protected void execute(Event event) {
+		// All sounds pattern wants explicitly defined master category
+		SoundCategory category = this.category.getOptionalSingle(event)
+			.orElse(allSounds ? null : SoundCategory.MASTER);
+		Player[] targets = players.getArray(event);
+		
+		if (allSounds) {
+			if (category == null) {
+				for (Player player : targets)
+					player.stopAllSounds();
 			} else {
-				if (SOUND_CATEGORIES_EXIST) {
-					for (Player p : players.getArray(e))
-						p.stopSound(soundEnum, (SoundCategory) category);
-				} else {
-					for (Player p : players.getArray(e))
-						p.stopSound(soundEnum);
+				for (Player player : targets)
+					player.stopSound(category);
+			}
+		} else {
+			for (String sound : sounds.getArray(event)) {
+				try {
+					Sound soundEnum = Sound.valueOf(sound.toUpperCase(Locale.ENGLISH));
+					for (Player player : targets)
+						player.stopSound(soundEnum, category);
+					
+					continue;
+				} catch (IllegalArgumentException ignored) {}
```

Comment:
I would keep a similar approach to how it was done before (using `Player#stopSound(String)`), right now the way you have it makes it so you can only use the enum names, and not the namespaced key of the sounds.
This also can be problematic if you're trying to stop custom sounds (sounds from texture packs) as enums are constant and would not have the namespaced key of the custom sound

Reply:
This also allows for namespaced keys to be used, look at line 117-124.

Reply:
Ah, right. I didn't see that mb